src/plugins/application_config/server/routes/index.ts

```diff
@@ -0,0 +1,162 @@
+      const client = getConfigurationClient(context.core.opensearch.client);
+
+      return await handleGetEntityConfig(client, request, response, logger);
+    }
+  );
+  router.post(
+    {
+      path: '/api/appconfig/{entity}',
+      validate: {
+        params: schema.object({
+          entity: schema.string(),
+        }),
+        body: schema.object({
+          newValue: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const client = getConfigurationClient(context.core.opensearch.client);
+
+      return await handleUpdateEntityConfig(client, request, response, logger);
+    }
+  );
+  router.delete(
+    {
+      path: '/api/appconfig/{entity}',
+      validate: {
+        params: schema.object({
+          entity: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const client = getConfigurationClient(context.core.opensearch.client);
+
+      return await handleDeleteEntityConfig(client, request, response, logger);
+    }
+  );
+}
+
+export async function handleGetEntityConfig(
+  client: ConfigurationClient,
+  request: OpenSearchDashboardsRequest,
+  response: OpenSearchDashboardsResponseFactory,
+  logger: Logger
+) {
+  try {
+    const result = await client.getEntityConfig(request.params.entity);
+    return response.ok({
+      body: {
+        value: result,
+      },
+    });
+  } catch (e) {
+    logger.error(e);
+    return errorResponse(response, e);
+  }
+}
+
+export async function handleUpdateEntityConfig(
+  client: ConfigurationClient,
+  request: OpenSearchDashboardsRequest,
+  response: OpenSearchDashboardsResponseFactory,
+  logger: Logger
+) {
+  try {
+    const result = await client.updateEntityConfig(request.params.entity, request.body.newValue);
+    return response.ok({
+      body: {
+        newValue: result,
+      },
+    });
+  } catch (e) {
+    logger.error(e);
+    return errorResponse(response, e);
+  }
+}
+
+export async function handleDeleteEntityConfig(
+  client: ConfigurationClient,
+  request: OpenSearchDashboardsRequest,
+  response: OpenSearchDashboardsResponseFactory,
+  logger: Logger
+) {
+  try {
+    const result = await client.deleteEntityConfig(request.params.entity);
+    return response.ok({
+      body: {
+        deletedEntity: result,
+      },
+    });
+  } catch (e) {
+    logger.error(e);
+    return errorResponse(response, e);
+  }
+}
+
+export async function handleGetConfig(
+  client: ConfigurationClient,
+  response: OpenSearchDashboardsResponseFactory,
+  logger: Logger
+) {
+  try {
+    const result = await client.getConfig();
+    return response.ok({
+      body: {
+        value: result,
+      },
+    });
+  } catch (e) {
+    logger.error(e);
+    return errorResponse(response, e);
+  }
+}
+
+function errorResponse(response: OpenSearchDashboardsResponseFactory, error: any) {
+  return response.customError({
+    statusCode: error?.statusCode || 500,
```

Comment:
I think appsec would not be happy about 500 error code. But it is just call out, not a blocker. you might want to check this when you do app sec review

Reply:
I will follow up on this during review. Thanks for the call out!